src/collection-view.js

```diff
@@ -151,6 +151,43 @@ Marionette.CollectionView = Marionette.AbstractView.extend({
     return this;
   },
 
+  // An efficient rendering used for filtering. Instead of modifying
+  // the whole DOM for the collection view, we are only adding or
+  // removing the related childrenViews.
+  setFilter: function(filter) {
+    if (!filter || this.filter === filter) {
+      return;
+    }
+    this.triggerMethod('before:filter', this);
+
+    this._resolveDeltasForFiltering(filter);
+
+    this.triggerMethod('filter', this);
+  },
+
+  // Calculate the deltas to remove/add the related childrenViews,
+  // so you don't need to rebuild the whole DOM.
+  _resolveDeltasForFiltering: function(filter) {
+    var previousModels = this._filteredSortedModels();
+    this.filter = filter;
+    var models = this._filteredSortedModels();
+    var currentIds = _.pluck(models, 'cid');
+    var modelMustBeShown;
+    // We resolve the deltas in a different way because we have
+    // to respect the sorting algorithm.
+    _.each(models, function(model, index) {
+      if (!this.children.findByModel(model)) {
+        this._onCollectionAdd(model, this.collection, {at:index});
```

Comment:
I'm not sure this will work with a custom view comparator. Can you please add a test to confirm?